### PR TITLE
[IOTDB-2474] Add progress logger when restarting IoTDB

### PR DESCRIPTION
**Pull request by HeimingZ**
Use scheduled logging service to log the recover progress.

### CODE REVIEW

File: server/src/assembly/resources/conf/iotdb-engine.properties [hunk 1]
```diff
@@ -319,6 +319,10 @@ timestamp_precision=ms
 # Datatype: int
 # virtual_storage_group_num = 1
 
+# the interval to log recover progress of each vsg when starting iotdb
+# Datatype: int
+# recovery_log_interval_in_ms=60000
```
Comment (JackieTien97) [inline server/src/assembly/resources/conf/iotdb-engine.properties:324]: The Default interval is a little too long, I think 5s will be better.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: server/src/assembly/resources/conf/iotdb-engine.properties (hunk 1 above)
Comment (qiaojialin) [reply server/src/assembly/resources/conf/iotdb-engine.properties:324]: He convinced me that 5s is ok.

Comment (qiaojialin) [inline server/src/assembly/resources/conf/iotdb-engine.properties:324]:
```suggestion
# recovery_log_interval_in_ms=5000
```

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java [hunk 2]
```diff
@@ -783,6 +783,9 @@
   /** the number of virtual storage groups per user-defined storage group */
   private int virtualStorageGroupNum = 1;
 
+  /** the interval to log recover progress of each vsg when starting iotdb */
+  private long recoveryLogIntervalInMs = 60_000L;
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBConfig.java:787]:
```suggestion
  private long recoveryLogIntervalInMs = 5_000L;
```

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/VirtualStorageGroupProcessor.java [hunk 3]
```diff
@@ -459,13 +447,47 @@ public void setReady(boolean ready) {
...
+
+    public void incrementRecoveredFilesNum() {
+      recoveredFilesNum++;
+      // check log only when 1% more files have been recovered
+      if (lastLogCheckFilesNum + filesNumLogCheckTrigger < recoveredFilesNum) {
+        lastLogCheckFilesNum = recoveredFilesNum;
+        // log only when log interval exceeds recovery log interval
+        if (lastLogTime + config.getRecoveryLogIntervalInMs() < System.currentTimeMillis()) {
+          logger.info(
+              "The virtual storage group {}[{}] is recovering {}%, please wait a moment.",
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/VirtualStorageGroupProcessor.java:479]:
```suggestion
              "The virtual storage group {}[{}] has recovered {}%, please wait a moment.",
```

**Pull request merged**